### PR TITLE
added undo functionality in event log

### PR DESCRIPTION
**Pull request by derekyyyan**
looks like this:
undo

### CODE REVIEW

**Review by lee7670 — COMMENTED** [review]
Maybe add a couple of comments, Looks good otherwise.

File: shared-data/default-theme/html/jsapi/global/eventlog.js [hunk 1]
```diff
@@ -259,3 +259,16 @@ $(document).ready(function () {
...
+  var event_id = $(this).data('event_id');
+  Mailpile.API.logs_events_undo_post({ event_id: event_id }, function(result) {
+    if (result.status === 'success') {
+      window.location.reload(true);
+    }
+    else {
+      alert("{{ _('Oops. Mailpile failed to complete your task.') }}");
+    }
+  });
+});
```
Comment (bigoju) [inline shared-data/default-theme/html/jsapi/global/eventlog.js:274]: Nice use of Mailpile undo API. LGTM !

File: shared-data/default-theme/html/logs/events/index.html [hunk 2]
```diff
@@ -28,20 +28,21 @@
        href="{{ U('/logs/events/?ui_expand=', event.event_id, '#', event.event_id) }}">
       {{ event.message }}
     </a>
+    {%- if event.data and event.data.undo %}
+    <a href="#"
+       title="{{ event.data.undo }}"
+       class="auto-modal eventlog-undo"
+       data-event_id="{{ event.event_id }}"
+       style="font-style: italic;
+              text-decoration: underline;">{{ _('Undo') }}</a>
```
Comment (bigoju) [inline shared-data/default-theme/html/logs/events/index.html:37]: Good job implementing text formatting to  expose Undo functionality to user viewing the logs!

**Comment by bigoju** [comment]
Nice LGTM.

**Pull request merged**